### PR TITLE
Create initial footer component

### PR DESCRIPTION
**Pull request by chillkang**
## Done
Created initial `footer` component

## QA
- Go to https://store-components-14.demos.haus/?path=/story/footer--page
- Check that there is a footer.

## Fixes
Fixes: https://warthogs.atlassian.net/browse/WD-1613

### CODE REVIEW

File: src/components/Footer/Footer.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+This is a [React](https://reactjs.org/) component for a store footer.
+
+### Props
+
+<ArgsTable of={Footer} />
+
+### Default
+
+<Footer copyrightYear={2023} companyName="Canonical Ltd." />
```
Comment (steverydz) [inline src/components/Footer/Footer.stories.mdx:20]: This needs to be:

```storybook
<Canvas>
  <Story
    name="Default"
    args={{
      copyrightYear: "2023",
      companyName: "Canonical Ltd."
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
```

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Row, Col, Strip } from "@canonical/react-components";
+
+export type Props = {
+  copyrightYear: number;
+  companyName: string;
+};
+
+const Footer: React.FC<Props> = ({ copyrightYear, companyName }) => {
```
Comment (steverydz) [inline src/components/Footer/Footer.tsx:9]: I don't think `companyName` will need to be a variable as it will always be "Canonical Ltd."

Comment (steverydz) [inline src/components/Footer/Footer.tsx:9]: We tend to use the `function` syntax for components, so in this case:

```js
function Footer({ copyRightYear }) {
```

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <>
```
Comment (steverydz) [inline src/components/Footer/Footer.tsx:11]: This doesn't need the fragment tag (`<></>`)

**Review by steverydz — CHANGES_REQUESTED** [review]

File: src/components/Footer/Footer.tsx (hunk 2 above)
Comment (steverydz) [reply src/components/Footer/Footer.tsx:9]: Come to think of it, the copyright year also shouldn't be a prop. As this component is JS we can use `new Date().getFullYear()` instead so it is always correct.

Comment (chillkang) [reply src/components/Footer/Footer.tsx:9]: I'll remove it :)

Comment (chillkang) [reply src/components/Footer/Footer.tsx]: I removed copyrightYear variable as well :)

**Review by steverydz — APPROVED** [review]
LGTM 👍

**Pull request merged**